lib/src/html/html_generator_instance.dart

```diff
@@ -89,76 +89,78 @@ class HtmlGeneratorInstance {
 
     generatePackage();
 
-    for (var lib in filterNonDocumented(_packageGraph.libraries)) {
-      generateLibrary(_packageGraph, lib);
-
-      for (var clazz in filterNonDocumented(lib.allClasses)) {
-        generateClass(_packageGraph, lib, clazz);
-
-        for (var constructor in filterNonDocumented(clazz.constructors)) {
-          if (!constructor.isCanonical) continue;
-          generateConstructor(_packageGraph, lib, clazz, constructor);
+    for (var package in _packageGraph.localPackages) {
+      for (var lib in filterNonDocumented(package.libraries)) {
+        generateLibrary(_packageGraph, lib);
+
+        for (var clazz in filterNonDocumented(lib.allClasses)) {
+          generateClass(_packageGraph, lib, clazz);
+
+          for (var constructor in filterNonDocumented(clazz.constructors)) {
```

Comment:
I see 7 things here w/ pretty much the same pattern.

Not for this PR, but wonder if this could be factored into a helper function so it's more readable – and less likely to have copy-paste errors

Reply:
Yep.

This can probably be done with a map of Type to Closure in html_generator_instance and helper methods in each ModelElement class to return documentable subitems.  That way you could crush this down into some nested expand calls.

Reply:
Though as I say that, while that's a pattern I've used in Python I've never actually tried to do that in Dart before.